### ISSUE TITLE
Bullet points starting with maths are not properly converted

### ISSUE THREAD

**Issue by tlienart** [tlienart/JuDoc.jl#45]
This kind of markdown:

```
* $f$ is a function that... 
* one can compute ...
```

leads to this kind of html

```html
<ul>
<li></li>
</ul>
\(f\) is in the set \(\Gamma_0(X)\) of convex functions on \(X\) that are <em>proper</em> and <em>lower semi-continuous</em>,</p>
<ul>
<li><p>one can compute a gradient or subgradient of </p>
</li>
</ul>
```

The second one is fine but the first one is not. This may have to do with the order in which things are done and eventually `md2html` is called. 

What it does:

* only parses a `*` then closes it as a standalone `<ul></ul>` with a single, empty, `<li></li>`
* then plug in the maths thing

**Comment by tlienart** [comment]
so this is part of a larger issue about context

what was happening:

```
* text
* (maths)
```

would be processed as

```
* text
* 
```
then 
```
(maths)
```
(an XBLOCK). 

Now working on this in branch context to try to make sure things are properly plugged in their context.

**Comment by tlienart** [comment]
this is fixed with the context add in cc6ae92424972fad3392b06247c19fc1c9511c1a
